compiler/ccgtypes.nim

```diff
@@ -123,10 +123,11 @@ const
 
 proc typeName(typ: PType): Rope =
   let typ = typ.skipTypes(irrelevantForBackend)
-  result = if typ.sym != nil and typ.kind in {tyObject, tyEnum}:
-             typ.sym.name.s.mangle.rope
-           else:
-             ~"TY"
+  result =
+    if typ.sym != nil and typ.kind in {tyObject, tyEnum}:
+      typ.sym.name.s.mangle.rope
+    else:
+      rope($typ.kind)
```

Comment:
This needs to be ``rope("TY" & $typ.kind)`` in order to not break the assumptions behind the name mangling rules, I think (could be wrong).